src/service/core/vmHandler.go

```diff
@@ -69,18 +69,29 @@ func (dm *KVMS) UnMapLabelIdentity(identity uint16, ewName string, labels []stri
 	}
 
 	for _, label := range labels {
+		deleted := false
 		identities := dm.MapLabelToIdentities[label]
-		for index, value := range dm.MapLabelToIdentities[label] {
-			if value == identity {
-				identities[index] = identities[len(identities)-1]
-				identities[len(identities)-1] = 0
-				identities = identities[:len(identities)-1]
+		if len(identities) > 1 {
```

Comment:
should this be `> 1` or `> 0`? Just confirming..

Reply:
No, it should be `> 1`. Because if there is just one identity associated with the label, then it means that when we delete the identity (VM), we should also remove the label from `dm.MapLabelToIdentities` and corresponding etcd path.